### PR TITLE
Nel docs

### LINKED ISSUE #321: `NonEmptyList` docs

### PR DESCRIPTION
**Pull request by aballano**
* Added Nel docs.
* Small fixes around
* Added Unit instances and changed Validated tests
* Some docs fixes.

Closes #321

### CODE REVIEW

File: kategory-docs/docs/docs/datatypes/nonemptylist/README.md [hunk 1]
```diff
@@ -6,4 +6,98 @@ permalink: /docs/datatypes/nonemptylist/
 
 ## NonEmptyList
 
-TODO
\ No newline at end of file
+NonEmptyList or Nel, is a list that, as the name suggests, is guaranteed to have at least one element.
+
+Let's explore it more in deep with a custom validation example:
```
Comment (pakoito) [inline kategory-docs/docs/docs/datatypes/nonemptylist/README.md]: in depth

File: kategory-docs/docs/docs/datatypes/nonemptylist/README.md [hunk 2]
```diff
@@ -6,4 +6,98 @@ permalink: /docs/datatypes/nonemptylist/
...
+            "no.at.com"
+    )
+            
+    emails.map { validateEmail(it) }
+            .reduce { acc, validated -> acc.combine(validated) }
+```
+
+As you can see we just obtain an Invalid case with all the errors combined, but they're not readable at all.
+
+### Validation with Nel
```
Comment (raulraja) [inline kategory-docs/docs/docs/datatypes/nonemptylist/README.md]: thius is actually not the common use case for `ValidatedNel` using `combineK` to actually accumulate errors on the left side when reducing. An example of that is here that is able to run a bunch of ops and determine which ones fails https://github.com/Guardiola31337/uiagesturegen/blob/pg-uiagesturegen-dsl/uiagesturegen/src/main/kotlin/com/pguardiola/uigesturegen/dsl/interpreter.kt#L36

File: kategory-docs/docs/docs/datatypes/nonemptylist/README.md [hunk 3]
```diff
@@ -6,4 +6,98 @@ permalink: /docs/datatypes/nonemptylist/
...
+
+NonEmptyList is highly connected with Validation, as it provides a nice and simple way of accumulating values for both the Valid or Invalid case. Let's see how it combines with it following the previous example:
+
+```kotlin:ank:silent
+import kategory.*
```
Comment (pakoito) [inline kategory-docs/docs/docs/datatypes/nonemptylist/README.md]: You don't need duplicate imports. The imports holds for the whole file, and it makes ank way way slower.

**Comment by raulraja** [comment]
@aballano you will have to merge your changes into some of the new docssections  we have added while this PR is still in Flux in order to record the videos. #713

**Comment by aballano** [comment]
Sorry folks, I don't think I'll have time in the near future to jump on this again and I don't want to block others from doing it, so I'll close it right away.

**Pull request closed without merging**